core/src/utilities/random.cc

```diff
@@ -735,4 +775,88 @@ namespace jiminy
             }
         };
     }
-}
\ No newline at end of file
+
+    HeightmapFunction unidirectionalRandomPerlinGround(
+        double wavelength, std::size_t numOctaves, double orientation, uint32_t seed)
+    {
+        const Eigen::Rotation2D<double> rot_mat(orientation);
+
+        // Constant for avoiding discontinuity at (0.0, 0.0)
+        const double offset = 100.0;
+
+        return [wavelength, numOctaves, rot_mat, offset, seed](
+                   const Eigen::Vector2d & pos, double & height, Eigen::Vector3d & normal) -> void
+        {
+            // Set seed of Random Perlin Process
+            PCG32 pcg32_generator = PCG32(seed);
```

Comment:
Move the instantiation of the Perlin process outside the lambda, and capture it (by copy) in the lambda.

Reply:
I am having trouble with capturing the Perlin process by copy since in the code, the copy is explicitly disabled:
```       
JIMINY_DISABLE_COPY(AbstractPerlinProcess)
```

Reply:
Try using move semantic then ! I guess it should work: `[generator=std::move(pcg32_generator)] (...) {...}`. Otherwise, I can implement the copy constructor, it should be fairly straightforward. It was just disable by simplicity.

Reply:
Ok thanks, I wasn't sure if I could remove this part or if it was on purpose. I have implemented the copy constructor if it works for you.